l2geth/accounts/abi/bind/backends/simulated.go

```diff
@@ -601,8 +601,9 @@ func (m callmsg) Gas() uint64          { return m.CallMsg.Gas }
 func (m callmsg) Value() *big.Int      { return m.CallMsg.Value }
 func (m callmsg) Data() []byte         { return m.CallMsg.Data }
 
```

Comment:
I think this is good, just wondering if there is a way to reduce the diff by not needing getters for the txmeta stuff longer term

Reply:
It might not be possible but would reduce a lot of diff

Reply:
Hmmm yeah I'm not sure, I'll take a look at that in one of my later PRs where I'll aim to minimize the diff as much as possible

Reply:
I think the way to do this is to have the blocknumber and timestamp live in the state